RecaptchaValidator/RecaptchaValidator.js

```diff
@@ -0,0 +1,88 @@
+const superagent = require('superagent');
+
+const GOOGLE_RECAPTCHA_SERVICE_HOST = 'https://www.google.com';
+const GOOGLE_RECAPTCHA_SERVICE_PATH = '/recaptcha/api/siteverify';
+const GOOGLE_RECAPTCHA_SERVICE_URL = GOOGLE_RECAPTCHA_SERVICE_HOST + GOOGLE_RECAPTCHA_SERVICE_PATH;
+
+const ERROR_MESSAGES = {
+    NO_CODE_PROVIDED: 'No recaptcha code provided. Please try again.',
+    INVALID_CODE_PROVIDED: 'Invalid/malformed recaptcha code provided. Please try again.',
+    DUPLICATE_OR_EXPIRED_CODE: 'The recaptcha code provided has expired or been previously used. Please try again.',
+    TRANSPORT_ERROR_ENCOUNTERED: 'An error was encountered validating your reCaptcha. Please try again.',
+    INTERNAL_ERROR_ENCOUNTERED: 'An error was encountered validating your reCaptcha. Please contact #wallet-support.',
+};
+
+const GOOGLE_API_ERROR_CODES = {
+    NO_CODE_PROVIDED: 'missing-input-response',
+    INVALID_CODE_PROVIDED: 'invalid-input-response',
+    DUPLICATE_OR_EXPIRED_CODE: 'timeout-or-duplicate',
+    MISSING_SECRET: 'missing-input-secret',
+    INVALID_SECRET: 'invalid-input-secret',
+    BAD_REQUEST: 'bad-request'
+};
+
+const ERROR_CODES = {
+    ...GOOGLE_API_ERROR_CODES,
+    TRANSPORT_ERROR: 'TRANSPORT_ERROR'
+};
+
+const RESPONSES_BY_ERROR_CODE = {
+    [ERROR_CODES.NO_CODE_PROVIDED]: { message: ERROR_MESSAGES.NO_CODE_PROVIDED, statusCode: 402 },
+    [ERROR_CODES.INVALID_CODE_PROVIDED]: { message: ERROR_MESSAGES.INVALID_CODE_PROVIDED, statusCode: 402 },
+    [ERROR_CODES.DUPLICATE_OR_EXPIRED_CODE]: { message: ERROR_MESSAGES.DUPLICATE_OR_EXPIRED_CODE, statusCode: 402 },
+    [ERROR_CODES.MISSING_SECRET]: { message: ERROR_MESSAGES.INTERNAL_ERROR_ENCOUNTERED, statusCode: 500 },
+    [ERROR_CODES.INVALID_SECRET]: { message: ERROR_MESSAGES.INTERNAL_ERROR_ENCOUNTERED, statusCode: 500 },
+    [ERROR_CODES.BAD_REQUEST]: { message: ERROR_MESSAGES.INTERNAL_ERROR_ENCOUNTERED, statusCode: 500 },
+
+    [ERROR_CODES.TRANSPORT_ERROR]: { message: ERROR_MESSAGES.TRANSPORT_ERROR_ENCOUNTERED, statusCode: 503 }
+};
+
+function getResponseFromErrorCode(errorCode) {
+    return RESPONSES_BY_ERROR_CODE[errorCode] || {
+        message: ERROR_MESSAGES.INTERNAL_ERROR_ENCOUNTERED,
+        statusCode: 500
+    };
+}
+
+class RecaptchaValidator {
+    constructor({
+        request = superagent,
+        RECAPTCHA_SECRET
+    }) {
+        this.request = request;
+        this.RECAPTCHA_SECRET = RECAPTCHA_SECRET;
+    }
+
+    // Documentation: https://developers.google.com/recaptcha/docs/verify
+    async validateRecaptchaCode(recaptchaCode, remoteIp) {
```

Comment:
nit: consider using a function instead of immutable object with one method

```
module.exports = (request, recaptchaSecret) => (recaptchaCode, remoteIp) => {
    // basically same code here
}
```

Reply:
I'll give this a try next time through -- generally, I try to avoid closure scope complexity as a default; I have found that as codebases get more complex, using closures to manage scope becomes harder to reason about, easier to introduce memory leaks, harder to debug, more complex to document (either with JSDoc or when adding types), and more difficult to refactor (especially using automated tools) than using statically defined objects.  Also, IDEs have a hard time with inferring about things that are passed in dynamically using thunks; I generally see much better code completion and intellisense when things are statically defined even if it's a little bit more verbose to define them that way